File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    
    for(int i = 0; i < n; i++) {
        string tmp = "";
        
        int num = arr1[i] | arr2[i];
        
        for(int j = 0; j < n; j++) {
            if(num & 1) 
                tmp = "#" + tmp;
            else 
                tmp = " " + tmp;
            
            num = num >> 1;
        }
        
        answer.push_back(tmp);
    }
    
    return answer;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for num1, num2 in zip(arr1, arr2):
        line_num = num1 | num2
        line_str = ""
        
        for _ in range(n):
            line_str = "#" + line_str if line_num % 2 == 1 else " " + line_str
            line_num = line_num // 2
        
        answer.append(line_str)
    
    return answer